src/AssetsPage/index.jsx

```diff
@@ -1,26 +1,37 @@
 import React from 'react';
+import statusMap from './statusMap.json';
 
 class AssetsPage extends React.Component {
-  static request() {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiConnectionStatus: 200,
+    };
+  }
+
+  request() {
     fetch('/api/assets/course-v1:edX+DemoX+Demo_Course/?page=0&page_size=50&sort=sort&asset_type=', {
       credentials: 'same-origin',
       headers: {
         Accept: 'application/json',
       },
     })
-      .then(data => data.json())
-      .catch(() => {
-        // console.log(error);
-      })
-      .then(() => {
-        // console.log(json);
+      .then((response) => {
+        if (response.status !== this.state.apiConnectionStatus) {
+          this.setState(() => ({ apiConnectionStatus: response.status }));
```

Comment:
FYI you can just do `this.setState({apiConnectionStatus: response.status});` here. It will shallow merge that object into the current state.